salt/_modules/metalk8s_monitoring.py

```diff
@@ -0,0 +1,173 @@
+"""Utiliy methods to interact with MetalK8s monitoring.
+"""
+
+from salt.exceptions import CommandExecutionError
+
+from datetime import datetime, timedelta
+
+MISSING_DEPS = []
+
+try:
+    import requests
+    from requests.packages.urllib3.exceptions import InsecureRequestWarning
+except ImportError:
+    MISSING_DEPS.append('requests')
+
+__virtualname__ = 'metalk8s_monitoring'
+
+CONTROL_PLANE_INGRESS_PORT = 8443
+
+
+def __virtual__():
+    if MISSING_DEPS:
+        error_msg = 'Missing dependencies: {}'.format(', '.join(MISSING_DEPS))
+        return False, error_msg
+
+    requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+    return __virtualname__
+
+
+def _requests_alertmanager_api(route, method='GET', **kwargs):
```

Comment:
I don't know for the PEP8 standard, is there anything about that ? I'm not sure. But we don't follow any standard for sure in our modules.

Reply:
None that I know of.

Reply:
Not sure if PEP8 says anything about this, as it's not really about formatting the code but more about organizing modules. If you want to make sure, feel free to read it and tell us :wink:

On the other hand, we could define our own standards for how to organize these modules, and indeed, where to put helper functions.
As a common practice, I usually try to define pieces of code before they are used. However, in Salt modules, it would probably make sense to have all the public methods first.

Reply:
FWIW, when the language allows it I usually organize the code is a top-down fashion (i.e the helpers tends to be further down or at the end of the file).

Rationale: I assume that you start reading a file from the top (to read the module doc for instance). Then you read down from here.
If you're only looking for high level info, you'll get what you want in the first functions.
If you need more details, you keep reading further into the file, to dive into the implementation details.

When looking for high level info, you don't have to read each function to know if it's an internal helper (and you need to skip it) or if it's an "interesting" function.


> On the other hand, we could define our own standards for how to organize these modules, and indeed, where to put helper functions.

This could go in our Python best practice, if we come up with something that please everyone.

Reply:
I've moved the helper method to the bottom of the file.